### PR TITLE
Feat/debug data warn spread ents

### PR DESCRIPTION
**Pull request by DuyguA**
## Description

Additions for warning users about entities in training data that cross sentence boundaries. An example:

```
I went there on [Friday. Aidan] came too.
```
Most probably this is an annotation mistake and not what is meant. 

### Types of change

Minor enhancement

## Checklist

- [x] I confirm that I have the right to submit this contribution under the project's MIT license.
- [x] I ran the tests, and all new and existing tests passed.
- [x] My changes don't require a change to the documentation, or if they do, I've added all required information.

### CODE REVIEW

**Review by polm — APPROVED** [review]
Looks good! It'll be great to finally catch this in `debug data`.

File: spacy/cli/debug_data.py [hunk 1]
```diff
@@ -242,12 +243,20 @@ def debug_data(
                     msg.warn(f"No examples for texts WITHOUT new label '{label}'")
                     has_no_neg_warning = True
 
+        if gold_train_data["boundary_cross_ents"]:
+            msg.warn(
+                f"{gold_train_data['boundary_cross_ents']} entity spans that cross sentence boundaries"
```
Comment (svlandeg) [inline spacy/cli/debug_data.py:248]:
```suggestion
                f"{gold_train_data['boundary_cross_ents']} entity span(s) crossing sentence boundaries"
```

File: spacy/cli/debug_data.py [hunk 2]
```diff
@@ -242,12 +243,20 @@ def debug_data(
...
+            has_boundary_cross_ents_warning = True
+
         if not has_low_data_warning:
             msg.good("Good amount of examples for all labels")
         if not has_no_neg_warning:
             msg.good("Examples without occurrences available for all labels")
         if not has_ws_ents_error:
             msg.good("No entities consisting of or starting/ending with whitespace")
+        if not has_boundary_cross_ents_warning:
+            msg.good("No entity spans crossing sentence boundaries")
```
Comment (svlandeg) [inline spacy/cli/debug_data.py:259]:
```suggestion
            msg.good("No entities crossing sentence boundaries")
```

**Review by svlandeg — APPROVED** [review]
Looks good! I just had a few nitpicking comments about rephrasing, I'll add those and then this can be merged :-)

**Pull request merged**